nix/setup.sh

```diff
@@ -71,8 +71,7 @@ fi
 
 # Check we have sudo capabilities
 echo "Testing sudo access"
-sudo sh -c 'echo $UID > /dev/null'
-if [[ $? -ne 0 ]] ; then
+if sudo bash -c 'echo $UID > /dev/null'; then
```

Comment:
You forgot to use the `$? -ne 0` comparison, so this check fails when the sudo access is present.